### PR TITLE
[WFLY-17160] Drop -jakarta suffix for final artifacts and use suffix for unconverted yet artifacts instead

### PR DESCRIPTION
**Pull request by jbliznak**
https://issues.redhat.com/browse/WFLY-17160

The main purpose of this commit is to avoid any more renaming happening in dependent projects while still allowing using source-transformation for modules not yet migrated to Jakarta namespaces.

The temporary transformed artifacts were using -jakarta suffix, however, as the name suggests, this was only a temporary solution and this artifact will no longer exist once the original one is transformed to Jakarta namespaces. This has undesired effect on any dependent project (eg. using explicit dependency managed by WF BOM) that would have to keep track of those changes and update on every such change.

The proposal of this change is doing it the other way around: use suffixed name for original untransformed artifacts and use original name for final transformed artifacts. That way even when the module get transformed we just drop usage of the old/intermediate artifact we no longer need while the "public API" remains the same for good.

The change was prepared by this script https://gist.github.com/jbliznak/a46b73355bb036125350dbad01dc036c
As these variable suggest, we can choose better suited names or strings once agreed on:
```
UNCONVERTED_ARTIFACT_ID_SUFFIX=-legacy-namespace
UNCONVERTED_ARTIFACT_NAME_SUFFIX=" (non-Jakarta Namespace)"
```

### CODE REVIEW

**Review by bstansberry — CHANGES_REQUESTED** [review]
The 'legacy-namespace' name is fine. It's going to be short-lived so I don't expect any other name will be better enough that it's worth thinking too hard.

My only request is that the '\$' be changed to '$'. It seems that it works even with that but I don't think we should leave cruft in those files.

File: bean-validation/pom.xml [hunk 1]
```diff
@@ -36,9 +36,9 @@
         <version>27.0.0.Final-SNAPSHOT</version>
     </parent>
 
-    <artifactId>wildfly-bean-validation</artifactId>
+    <artifactId>wildfly-bean-validation-legacy-namespace</artifactId>
 
-    <name>WildFly: Bean Validation</name>
+    <name>WildFly: Bean Validation (non-Jakarta Namespace)</name>
```
[the remark this answers is not in the document]
Comment (jbliznak) [reply bean-validation/pom.xml:41]: my motivation was to make it visible even in logs of build reactor that there is still something to do but I understand your POV too, no problem to leave that out

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/eclipse/persistence/main/module.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
     </properties>
 
     <resources>
-        <artifact name="\${org.wildfly:jipijapa-eclipselink@module.jakarta.suffix@}"/>
+        <artifact name="\${org.wildfly:jipijapa-eclipselink}"/>
```
[the remark this answers is not in the document]
Comment (jbliznak) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/eclipse/persistence/main/module.xml:32]: this is my mistake, the backslash should be removed, will fix

[the remark this answers is not in the document]
Comment (jbliznak) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/eclipse/persistence/main/module.xml]: fixed

**Review by bstansberry — APPROVED** [review]

**Review by scottmarlow — APPROVED** [review]

**Comment by bstansberry** [comment]
I just merged main into this branch to resolve a conflict introduced by merging a different PR into main.

**Comment by bstansberry** [comment]
Thanks, @jbliznak!

**Pull request merged**